host-controller/src/main/java/org/jboss/as/domain/controller/operations/deployment/ServerGroupDeploymentReplaceHandler.java

```diff
@@ -92,7 +92,8 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
         final List<ContentReference> locallyAddedReferences = new LinkedList<ContentReference>();
         for (ModelNode content : deployment.require(CONTENT).asList()) {
             if ((content.hasDefined(HASH))) {
-                ContentReference reference = ModelContentReference.fromModelAddress(address, content.require(HASH).asBytes());
+                final PathAddress referenceAddress = address.append(deploymentPath);
```

Comment:
the `referenceAddress` can be instantiated once outside of the `for` loop as it does not use anything from the content iteration.

Reply:
Fixed, thanks @jmesnil